base/ntuple.jl

```diff
@@ -91,3 +91,51 @@ end
         (t..., fill(val, N-M)...)
     end
 end
+
+# Specializations for copyto! of various `NTuple`s
+@noinline function check_inds_compatible(dest::AbstractArray, src::Tuple)
+    destinds = LinearIndices(dest)
+    idf, isf = first(destinds), firstindex(src)
+    Δi = idf - isf
+    (checkbounds(Bool, destinds, isf+Δi) & checkbounds(Bool, destinds, length(src)+Δi)) ||
+        throw(BoundsError(dest, srcinds))
```

Comment:
This method seems more complicated than it needs to be, and I don't really understand why it's not just a length check.

Also:

```
ERROR: UndefVarError: srcinds not defined
Stacktrace:
 [1] check_inds_compatible(dest::OffsetVector{Float64, Vector{Float64}}, src::NTuple{6, Int64})
   @ Main ./REPL[29]:5
```

Reply:
Yeah, should probably just be a length check.